src/services/api.ts

```diff
@@ -0,0 +1,62 @@
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { client } from '../utils/fetchClient';
+
+export const getUsers = (
+  setIsErrorShowing: (isShowing: boolean) => void,
+  setUsersFromServer: (users: User[] | null) => void,
+) => {
+  setIsErrorShowing(false);
+
+  return client.get<User[]>('/users')
+    .then(setUsersFromServer)
+    .catch(() => setIsErrorShowing(true));
+};
+
+export const getPosts = async (
+  userId: number,
+) => {
+  try {
+    return await client.get<Post[]>(`/posts?userId=${userId}`);
+  } catch {
+    throw new Error('Couldn\'t fetch posts');
+  }
+};
```

Comment:
```suggestion
export const getPosts = (
  userId: number,
) => {
  return client.get<Post[]>(`/posts?userId=${userId}`);
};
```
I meant like this, you don't need to pass set state functions and handle responses here